hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/TokenUpdateHandler.java

```diff
@@ -434,6 +401,107 @@ private void updateKeys(
+            }
+        }
+
+        if (op.hasTreasury()) {
+            context.requireKeyOrThrow(op.treasuryOrThrow(), INVALID_ACCOUNT_ID);
+        }
+        if (op.hasAutoRenewAccount()) {
+            context.requireKeyOrThrow(op.autoRenewAccountOrThrow(), INVALID_AUTORENEW_ACCOUNT);
+        }
+
+        if (originalToken.hasAdminKey()) {
+            // if we update any of the low priority keys, we should allow either admin key
+            // or the respective low priority key to sign.
+            if (noOtherFieldThanLowPriorityKeyOrMetadataWillBeUpdated(op)) {
+                List<Key> lowPriorityKeys = new ArrayList<>();
+                addRequiredLowPrioritySigner(
+                        lowPriorityKeys, originalToken.hasWipeKey(), originalToken.wipeKey(), op.hasWipeKey());
+                addRequiredLowPrioritySigner(
+                        lowPriorityKeys, originalToken.hasKycKey(), originalToken.kycKey(), op.hasKycKey());
+                addRequiredLowPrioritySigner(
+                        lowPriorityKeys, originalToken.hasSupplyKey(), originalToken.supplyKey(), op.hasSupplyKey());
+                addRequiredLowPrioritySigner(
+                        lowPriorityKeys, originalToken.hasFreezeKey(), originalToken.freezeKey(), op.hasFreezeKey());
+                addRequiredLowPrioritySigner(
+                        lowPriorityKeys,
+                        originalToken.hasFeeScheduleKey(),
+                        originalToken.feeScheduleKey(),
+                        op.hasFeeScheduleKey());
+                addRequiredLowPrioritySigner(
+                        lowPriorityKeys, originalToken.hasPauseKey(), originalToken.pauseKey(), op.hasPauseKey());
+                addRequiredLowPrioritySigner(
+                        lowPriorityKeys,
+                        originalToken.hasMetadataKey(),
+                        originalToken.metadataKey(),
+                        op.hasMetadataKey());
+                addRequiredLowPrioritySigner(
+                        lowPriorityKeys, originalToken.hasMetadataKey(), originalToken.metadataKey(), op.hasMetadata());
+
+                Key requiredKey;
+                if (lowPriorityKeys.isEmpty()) {
+                    requiredKey = originalToken.adminKey();
+                } else {
+                    final Key lowPriorityKeyList = Key.newBuilder()
+                            .keyList(KeyList.newBuilder().keys(lowPriorityKeys).build())
+                            .build();
+                    final List<Key> keysRequired = List.of(lowPriorityKeyList, originalToken.adminKey());
+                    // with this we will require either admin key signature or all low priority keys signatures
+                    // (we can update several low priority keys with one TokenUpdate, so we will require all of these
+                    // keys
+                    // to sign)
+                    requiredKey = Key.newBuilder()
```

Comment:
If I read this correctly, one of all the keys should sign. What happens if we are updating several keys in same update transaction ? This should just take adminKey or all the keys updated to sign the transaction

Reply:
Here we have `lowPriorityKeys` list of low priority keys to be updated. After that if the list is not empty, we create `lowPriorityKeyList` KeyList (N/N signers) and only then we create the `Threshold` key (1/2 signers) which requires either all keys from lowPriorityKeyList to sign or only the admin key to sign.

In that way, if we have 2 or more low prio keys to be updated we either require those low prio keys to sign or only the admin key signature